src/pool-cl/libraries/CLPool.sol

```diff
@@ -216,24 +212,33 @@ library CLPool {
             revert InvalidSqrtPriceLimit(slot0Start.sqrtPriceX96, sqrtPriceLimitX96);
         }
 
-        SwapCache memory cache = SwapCache({
-            liquidityStart: self.liquidity,
-            /// @dev 8 bits for protocol swap fee instead of 4 bits in v3
-            protocolFee: zeroForOne ? uint8(slot0Start.protocolFee % 256) : uint8(slot0Start.protocolFee >> 8)
-        });
-
+        // liquidity at the beginning of the swap
+        uint128 liquidityStart = self.liquidity;
         bool exactInput = params.amountSpecified > 0;
 
-        state = SwapState({
-            amountSpecifiedRemaining: params.amountSpecified,
-            amountCalculated: 0,
-            sqrtPriceX96: slot0Start.sqrtPriceX96,
-            tick: slot0Start.tick,
-            swapFee: slot0Start.swapFee,
-            feeGrowthGlobalX128: zeroForOne ? self.feeGrowthGlobal0X128 : self.feeGrowthGlobal1X128,
-            protocolFee: 0,
-            liquidity: cache.liquidityStart
-        });
+        {
+            uint16 protocolFee = params.zeroForOne
```

Comment:
(optional --- maybe can create as issue tracker if keen)
been seeing this `if` condition quite a bit. thinking if we might want to make our code cleaner.

eg.
```
uint16 protocolFee = slot0Start.protocolFee.getProtocolFee(param.zeroForOne);
```

Reply:
https://github.com/pancakeswap/pancake-v4-core/issues/40